doc: openthread: Describe nRF radio implementation (#23129)
nrfconnect/sdk-nrf: samples/openthread/cli/README.rst

If used alone, the sample allows you to test the network status.
It is recommended to use at least two development kits running the same sample for testing the communication.

Application architecture options
================================

.. ot_cli_sample_architecture_options_start

> The sample supports two application architecture options depending on the usage of the Zephyr networking layer.

Reviewer: ```suggestion
The sample supports two application architecture options depending on if you are using the Zephyr networking layer.
```

* Direct IEEE 802.15.4 radio integration with OpenThread Stack (default)
* Integration with Zephyr networking layer

To learn more about the differences between the two architectures, see the :ref:`openthread_stack_architecture` page.
To learn how to switch between the two architectures, see the :ref:`ug_thread_configuring_basic` user guide.

> Additionally, you can use the ``l2`` snippet to switch between the two architectures, see the :ref:`ot_cli_sample_activating_variants` section.

Reviewer: ```suggestion
Additionally, you can use the ``l2`` snippet to switch between the two architectures. 
See the :ref:`ot_cli_sample_activating_variants` section for details.
```
